mlflow/utils/autologging_utils.py

```diff
@@ -260,3 +292,409 @@ def batch_metrics_logger(run_id):
     batch_metrics_logger = BatchMetricsLogger(run_id)
     yield batch_metrics_logger
     batch_metrics_logger.flush()
+
+
+def autologging_integration(name):
+    """
+    **All autologging integrations should be decorated with this wrapper.**
+
+    Wraps an autologging function in order to store its configuration arguments. This enables
+    patch functions to broadly obey certain configurations (e.g., disable=True) without
+    requiring specific logic to be present in each autologging integration.
+    """
+
+    AUTOLOGGING_INTEGRATIONS[name] = {}
+
+    def wrapper(_autolog):
+        def autolog(**kwargs):
+            AUTOLOGGING_INTEGRATIONS[name] = kwargs
+            _autolog(**kwargs)
```

Comment:
```suggestion
    def wrapper(_autolog):
        def autolog(*args, **kwargs):
            AUTOLOGGING_INTEGRATIONS[name] = kwargs
            _autolog(*args, **kwargs)
```

do we need `*args` here?

Reply:
or should we disallow passing positional arguments to `mlflow.<flavor name>.autolog` using [the keyword-only-arguments syntax](https://www.python.org/dev/peps/pep-3102/#:~:text=Keyword%2Donly%20arguments%20are%20not,therefore%20'required%20keyword'%20arguments.)?

Reply:
I've disallowed passing positional arguments by introducing validation to `autologging_integration`. Arguments must either specify a default value or must be of type `KEYWORD_ONLY` (as determined by `inspect.signature`).

Reply:
@harupy Based on our offline discussion regarding backwards compatibility for supplying positional arguments to `autolog()` functions, I've introduced support for `args`. I've also added test coverage to ensure that args are reflected in the conf and are forwarded by the `autologging_integration` wrapper to the developer-specified `autolog()` function.